beacon/beacon.go

```diff
@@ -92,34 +92,80 @@ func parseInput(r *http.Request) (*variants.Query, error) {
 		var query variants.Query
 		query.RefName = r.FormValue("chromosome")
 		query.Allele = r.FormValue("allele")
-
-		coord, err := getFormValueInt(r, "coordinate")
-		if err != nil {
-			return nil, fmt.Errorf("parsing coordinate: %v", err)
+		if err := parseFormCoordinates(r, &query); err != nil {
+			return nil, fmt.Errorf("parsing referenceBases: %v", err)
 		}
-		query.Coord = coord
-
 		return &query, nil
 	case "POST":
 		var params struct {
-			RefName string `json:"chromosome"`
-			Allele  string `json:"allele"`
-			Coord   *int64 `json:"coordinate"`
+			RefName  string `json:"chromosome"`
+			Allele   string `json:"allele"`
+			Start    *int64 `json:"start"`
+			End      *int64 `json:"end"`
+			StartMin *int64 `json:"startMin"`
+			StartMax *int64 ` json:"startMax"`
+			EndMin   *int64 `json:"endMin"`
+			EndMax   *int64 `json:"endMax"`
 		}
 		body, _ := ioutil.ReadAll(r.Body)
 		if err := json.Unmarshal(body, &params); err != nil {
 			return nil, fmt.Errorf("decoding request body: %v", err)
 		}
+
 		return &variants.Query{
-			RefName: params.RefName,
-			Allele:  params.Allele,
-			Coord:   params.Coord,
+			RefName:  params.RefName,
+			Allele:   params.Allele,
+			Start:    params.Start,
+			End:      params.End,
+			StartMin: params.StartMin,
+			StartMax: params.StartMax,
+			EndMin:   params.EndMin,
+			EndMax:   params.EndMax,
 		}, nil
 	default:
 		return nil, errors.New(fmt.Sprintf("HTTP method %s not supported", r.Method))
 	}
 }
 
+func parseFormCoordinates(r *http.Request, params *variants.Query) error {
+	start, err := getFormValueInt(r, "start")
```

Comment:
Since these are all the same, I wonder about

fields := map[string]*int32{
  "start": &params.Start,
  ....
}

for name, field := range fields {
  v, err := getFormValue(r, name)
  if err != nil {
    return fmt.Errorf("parsing %s: %v", name, err)
  }
  *field = v
}

Reply:
that's much cleaner, done.